src/treetensornetworks/abstracttreetensornetwork.jl

```diff
@@ -420,3 +420,25 @@ function inner(
   end
   return O[]
 end
+
+function expect(
+  operator::String,
+  state::AbstractTTN;
+  vertices=vertices(state),
+  root_vertex=default_root_vertex(siteinds(state)), #ToDo: verify that this is a sane default
+)
+  # ToDo: for performance it may be beneficial to also implement expect! and change the orthogonality center in place
+  # assuming that the next algorithmic step can make use of the orthogonality center being moved to a different vertex
+  sites = siteinds(state)
+  ordered_vertices = reverse(post_order_dfs_vertices(sites, root_vertex)) #ToDo: Verify that this is indeed the correct order for performance
+  ElT = promote_itensor_eltype(state)
```

Comment:
I think you can use `scalartype(state)` here (same thing with a shorter name, and more generic).

Reply:
Another thing is that the operator could have a different element type than the state, so this code will fail in that case. We could build the operator on each site in `vertices` ahead of time, store that in a dictionary, and then extract the element type of the operator, and determine the output element type by promoting the operator element type and the state element type to a common element type.

Reply:
I was going to store the expectation values in an untyped (for the value) dictionary and determine the element type after the loop and create a new typed dictionary from the values and keys of the untyped one.
Should I pay attention to replacing usage of `psi` `operator` etc throughout the whole file (and similarly reflect style changes in src/expect.jl)?

Reply:
That seems like a good approach, you can use this pattern for that:
```julia
julia> d = Dictionary(["a", "b", "c"], undef)
3-element Dictionary{String, Any}
 "a" │ #undef
 "b" │ #undef
 "c" │ #undef

julia> d["a"] = 1.0
1.0

julia> d["b"] = 2.0
2.0

julia> d["c"] = 3.0
3.0

julia> mapreduce(typeof, promote_type, d).(d)
3-element Dictionary{String, Float64}
 "a" │ 1.0
 "b" │ 2.0
 "c" │ 3.0
```
I think in this PR we shouldn't change the style of the entire file, its generally better to keep PRs more narrowly focused in scope (either add a new feature and use the best style within that new code, or make a PR that improves the style across a lot of code).